multibody/multibody_solvers.h

```diff
@@ -52,7 +52,7 @@ class MultibodyProgram : public drake::solvers::MathematicalProgram {
   drake::solvers::Binding<drake::solvers::Constraint> AddKinematicConstraint(
       const KinematicEvaluatorSet<T>& evaluators,
       const drake::solvers::VectorXDecisionVariable& q,
-      std::shared_ptr<drake::systems::Context<T>> local_context);
+      drake::systems::Context<T>* local_context);
```

Comment:
I don't see `AddKinematicConstraint(evaluators, q, local_context)` being used anywhere in dairlib. Are we going to use the local_context version of AddXXXConstraint?

Reply:
I think it's important to have it as an option, in case multiple constraint sets are added that correspond to multiple state decision variables (to avoid dirtying the cache when going back and forth between variable sets).